libs/vm/src/analyser.cpp

```diff
@@ -925,6 +963,11 @@ void Analyser::BuildFreeFunctionDefinition(BlockNodePtr const &function_definiti
   ExpressionNodePtr function_name_node =
       ConvertToExpressionNodePtr(function_definition_node->children[1]);
   std::string const &function_name = function_name_node->text;
+  if (num_free_functions_++ >= MAX_FREE_FUNCTIONS)
+  {
+    throw FatalErrorException(filename_, function_name_node->line,
```

Comment:
Are these exceptions caught by the top level compile function?

Reply:
They are caught inside Analyse(), they can't escape that function.

Reply:
It means fatal error in the sense of fatal semantic error, not fatal error for the compiler itself.